src/tests/payment/test_stripe_webhook.py

```diff
@@ -0,0 +1,371 @@
+    ):
+        """
+        Complete test of the `checkout.session.completed`
+        function.
+        Checks that `CustomerDetail` and `Subscription`
+        are correctly created or updated.
+        """
+        sig_header = generate_stripe_signature(payload, STRIPE_WEBHOOK_SECRET)
+        mock_construct_event.return_value = payload
+
+        payload["data"]["object"]["metadata"]["user_id"] = user.id
+        payload["data"]["object"]["id"] = "sub_test_id"
+
+        response = client.post(
+            reverse_lazy("v1:payment:stripe-webhook"),
+            data=json.dumps(payload),
+            content_type="application/json",
+            HTTP_STRIPE_SIGNATURE=sig_header,
+        )
+
+        assert response.status_code == 200
+
+        customer_detail = CustomerDetail.objects.get(user=user)
+        assert (
+            customer_detail.stripe_customer_id == payload["data"]["object"]["customer"]
+        )
+        assert (
+            customer_detail.city
+            == payload["data"]["object"]["customer_details"]["address"]["city"]
+        )
+        assert (
+            customer_detail.country
+            == payload["data"]["object"]["customer_details"]["address"]["country"]
+        )
+        assert (
+            customer_detail.address
+            == payload["data"]["object"]["customer_details"]["address"]["line1"]
+        )
+        assert (
+            customer_detail.postal_code
+            == payload["data"]["object"]["customer_details"]["address"]["postal_code"]
+        )
+        assert (
+            customer_detail.email
+            == payload["data"]["object"]["customer_details"]["email"]
+        )
+
+        subscription = Subscription.objects.get(user=user)
+        assert subscription.stripe_subscription_id == payload["data"]["object"]["id"]
+        assert subscription.status == payload["data"]["object"]["status"]
+        assert (
+            subscription.payment_status == payload["data"]["object"]["payment_status"]
+        )
+        assert (
+            subscription.product_name
+            == payload["data"]["object"]["metadata"]["product_name"]
+        )
+        assert subscription.total_price == payload["data"]["object"][
+            "amount_total"
+        ] / Decimal(100)
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_checkout_session_completed_user_does_not_exist(
+        self, mock_construct_event, client, payload
+    ):
+        mock_construct_event.return_value = payload
+        payload["data"]["object"]["metadata"]["user_id"] = "9999"
+
+        sig_header = generate_stripe_signature(payload, STRIPE_WEBHOOK_SECRET)
+        response = client.post(
+            reverse_lazy("v1:payment:stripe-webhook"),
+            data=json.dumps(payload),
+            content_type="application/json",
+            HTTP_STRIPE_SIGNATURE=sig_header,
+        )
+        assert response.status_code == 400
+        # assert response.content == b"User does not exist" ToDo
```

Comment:
I put a little not so that we don't forget to address that one

Reply:
okay